sklearn/tests/test_discriminant_analysis.py

```diff
@@ -525,23 +541,32 @@ def test_qda_store_covariance():
 def test_qda_regularization():
     # the default is reg_param=0. and will cause issues
     # when there is a constant variable
+    # UserWarning will raised due to the constant variable
+    # Runtime warning will be raised due to unregularized
+    # constant variable in the covariance matrix.
     clf = QuadraticDiscriminantAnalysis()
-    with ignore_warnings():
-        y_pred = clf.fit(X2, y6).predict(X2)
+    with (pytest.warns(UserWarning) and
+          pytest.warns(RuntimeWarning)):
```

Comment:
Nice, I did not know you could do that with the `and` operator on this context manager!

Reply:
Could you please make

```python
import re

...

    constant_feature_msg = re.escape(
        "The expected message..."
    )
    covariance_msg = re.escape(
        "The other expected message..."
    )
    with (pytest.warns(UserWarning, match=constant_feature_msg) and
          pytest.warns(RuntimeWarning, match=covariance_msg)):
```

That would make it clearer about what's going on in this test (assuming the warning messages are informative).

Reply:
The message for the constant feature is very informative, the error for discriminant could be better, still included as shown here.